src/style/fog.js

```diff
@@ -29,16 +34,77 @@ const TRANSITION_SUFFIX = '-transition';
 export const FOG_PITCH_START = 55;
 export const FOG_PITCH_END = 65;
 
+export class FogSampler {
+    properties: ?PossiblyEvaluated<Props>;
+
+    // As defined in _prelude_fog.fragment.glsl#fog_opacity
+    getFogOpacity(depth: number, pitch: number): number {
+        if (!this.properties) { return 0.0; }
+
+        const props = this.properties;
+        const range = props.get('range');
+        const fogOpacity = props.get('opacity') * smoothstep(FOG_PITCH_START, FOG_PITCH_END, pitch);
```

Comment:
So forward compatibility looks like removing the smoothstep, and `props.get('opacity')` would have the smoothstep expression using pitch baked in as a default? Presumably a cubic bezier which may not be *exactly identical* but is within tolerance? https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#interpolate

Reply:
Yes, I would imagine something like this for a similar behavior:
```
opacity: ['interpolate', ['cubic-bezier', ...], ['pitch'], 55, 65]
```